src/main/java/jenkins/plugins/publish_over_ssh/BapSshHostConfiguration.java

```diff
@@ -156,7 +163,42 @@ public void setEncryptedPassword(final String encryptedPassword) {
         this.keyInfo.setPassphrase(encryptedPassword);
     }
 
-    public String getKeyPath() { return keyInfo.getKeyPath(); }
+
+    public Pattern getRegex() {
+		return regex;
+	}
+    
+    /**
+     * Sets regexValue and regex. Workaround for Jelly
+     * @param regexValue
+     */
+    @DataBoundSetter
+	public void setRegexValue(String regexValue) {
+    	this.regexValue=regexValue;
+        	try{
+        		regex= Pattern.compile(regexValue);
+        	} catch (PatternSyntaxException e){
+        		useRegex=false;
+        	}
+        	if(regexValue == null || Pattern.matches("\\u0008*",regexValue)){
+        		useRegex=false;
+        	}
+	}
+    
+    public String getRegexValue(){
+    	return regexValue;
+    }
+   
+    @DataBoundSetter
+    public void setUseRegex(boolean useRegex){
```

Comment:
Maybe `@Restricted(NoExternalUse.class)` if you want to prevent it's exposure to public API